meilisearch/models/index.py

```diff
@@ -29,6 +32,35 @@ def __iter__(self) -> Iterator:
 
 class Faceting(CamelBase):
     max_values_per_facet: int
+    sort_facet_values_by: Optional[Dict[str, str]] = None
+
+    if is_pydantic_2():
+
+        @pydantic.field_validator("sort_facet_values_by")  # type: ignore[attr-defined]
+        @classmethod
+        def validate_facet_order(cls, val: Optional[Dict[str, str]]) -> Optional[Dict[str, str]]:
+            if not val:  # pragma: no cover
+                return None
+
+            for _, value in val.items():
+                if value not in ("alpha", "count"):
+                    raise ValueError('facet_order must be either "alpha" or "count"')
+
+            return val
+
+    else:  # pragma: no cover
+
+        @pydantic.validator("sort_facet_values_by")  # type: ignore[attr-defined]
+        @classmethod
+        def validate_facet_order(cls, val: Optional[Dict[str, str]]) -> Optional[Dict[str, str]]:
+            if not val:
+                return None
+
+            for _, value in val.items():
+                if value not in ("alpha", "count"):
+                    raise ValueError('facet_order must be either "alpha" or "count"')
+
+            return val
```

Comment:
Since we are only using this value to return what Meilisearch returns and not send any values in this SDK we could remove this validation. Whatever you would rather do is good with me, just let me know if you want me to remove it.

Reply:
Hum, for me it's not up to the SDK to check the validation. Meilisearch will already send an error if you don't write a valid value. If Meilisearch adds a new value, we will have to update this part of the code, and the SDK could not be used directly without an implementation on our side, which is too bad.